src/indicators/drought/wrsi.py

```diff
@@ -0,0 +1,309 @@
+                download_wrsi(
+                    region=region,
+                    start_date=[max_year, max_dekad],
+                    end_date=[end_year, end_dekad],
+                    redownload=True,
+                )
+    else:
+        while not (
+            i_year > end_year or (i_year == end_year and i_dekad > i_dekad)
+        ):
+            _download_wrsi(
+                year=i_year, dekad=i_dekad, region=region, raw_dir=raw_dir
+            )
+            i_dekad += 1
+            if i_dekad > 33:
+                i_year += 1
+                i_dekad = 13
+
+
+_PROCESSED_FILENAME = "wrsi_{region}_{type}.nc"
+
+_PROCESSED_DIR = Path(
+    os.getenv("AA_DATA_DIR"), "public", "processed", "glb", "wrsi"
+)
+
+
+def _get_processed_path(region, type):
+    _PROCESSED_DIR.mkdir(parents=True, exist_ok=True)
+    return Path(
+        _PROCESSED_DIR,
+        _PROCESSED_FILENAME.format(region=region, type=type),
+    )
+
+
+# defined here: https://gis.stackexchange.com/questions/
+# 177447/defining-the-correct-aea-proj4string-for-
+# fewsnet-rainfall-data-southern-africa
+_WRSI_CRS = CRS.from_proj4(
+    "+proj=aea +lat_1=-19.0 +lat_2=21.0 +lat_0=1.0 "
+    "+lon_0=20 +x_0=0 +y_0=0 +ellps=clrk66 +units=m +no_defs"
+)
+
+
+def process_wrsi(region, type):
+    if region not in _VALID_REGION:
+        raise ValueError("`region` must be one of 'cropland' or 'rangeland'.")
+
+    raw_dir = _get_raw_dir(region)
+
+    types = {"current": "do", "extended": "eo", "anomaly": "er"}
+
+    if type in types.keys():
+        fp_type = types[type]
+    else:
+        raise ValueError(
+            "`type` must be one of 'current', 'extended', or 'anomaly'."
+        )
+
+    processed_path = _get_processed_path(region, type)
+
+    def _load_raw(fp):
+        arr = xr.open_rasterio(Path(raw_dir, fp))
+        dt = [_fp_date(fp)]
+
+        arr = arr.expand_dims(time=dt)
+        # round x dimension due to issues with
+        # concat and tiny numerical differences
+        arr = arr.assign_coords({"x": np.round(arr.x.values, 7)})
+        return arr
+
+    arrays = [
+        _load_raw(filename) for filename in raw_dir.glob(fp_type + "*.tif")
+    ]
+
+    arrays_merged = xr.concat(arrays, "time").sortby("time")
+
+    # change missing values from 0 to nan
+    arrays_merged = arrays_merged.where(arrays_merged.values > 0)
+    arrays_merged.attrs["_FillValue"] = np.NaN
+
+    # convert to dataset
+    ad = arrays_merged.to_dataset("band")
+    ad = ad.rename({1: "wrsi"})
+
+    # saving file
+    processed_path.unlink(missing_ok=True)
+
+    ad.to_netcdf(processed_path)
+
+    return ad
+
+
+def load_wrsi(region, type):
+    processed_path = _get_processed_path(region, type)
+    a = xr.open_dataset(processed_path).to_array("wrsi")
+    a = a.squeeze(drop=True)
+    a.rio.write_crs(_WRSI_CRS, inplace=True)
+    a = a.rio.reproject("EPSG:4326")
+    return a
+
+
+def filter_wrsi(da, dekad):
+    dt = _dekad_to_date(2001, dekad)
```

Comment:
aha so here you select the data from one dekad across all years? Would another option to have an extra coordinate named `dekad` (which is dependent on time) such that you can just filter on dekad. 

If you never use the `time` directly it could also be an option to instead have a `year` and `dekad` dimension but can imagine it is also easy to have them combined into one.

Reply:
Yeah, I have added as a TODO. The reason it was troublesome before, cause I tried, was beacuse then you have multiple dimensions and then I was using rioxarray and that was problematic. Then i switched it all up, a mess. Something to investigate soon, this great!